wandb/sdk/service/grpc_service.py

```diff
@@ -0,0 +1,397 @@
+        return result
+
+    def ServerStatus(  # noqa: N802
+        self,
+        request: wandb_server_pb2.ServerStatusRequest,
+        context: grpc.ServicerContext,
+    ) -> wandb_server_pb2.ServerStatusResponse:
+        assert self._backend and self._backend._interface
+        result = wandb_server_pb2.ServerStatusResponse()
+        return result
+
+
+# TODO(jhr): this should be merged with code in backend/backend.py ensure launched
+class GrpcBackend:
+    _interface: interface.BackendSender
+    _settings: Dict[str, Any]
+    _process: multiprocessing.process.BaseProcess
+    _record_q: "queue.Queue[pb.Record]"
+    _result_q: "queue.Queue[pb.Result]"
+    _monitor_pid: Optional[int]
+
+    def __init__(self, pid: int = None, debug: bool = False) -> None:
+        self._done = False
+        self._record_q = multiprocessing.Queue()
+        self._result_q = multiprocessing.Queue()
+        self._process = multiprocessing.current_process()
+        self._settings = self._make_settings()
+        self._monitor_pid = pid
+        self._debug = debug
+
+        if debug:
+            self._settings["log_internal"] = None
+
+        self._interface = wandb.wandb_sdk.interface.interface.BackendSender(
+            record_q=self._record_q,
+            result_q=self._result_q,
+            process=self._process,
+            process_check=False,
+        )
+
+    def _make_settings(self) -> Dict[str, Any]:
+        log_level = logging.DEBUG
+        start_time = time.time()
+        start_datetime = datetime.datetime.now()
+        timespec = datetime.datetime.strftime(start_datetime, "%Y%m%d_%H%M%S")
+
+        wandb_dir = "wandb"
+        pid = os.getpid()
+        run_path = "run-{}-{}-server".format(timespec, pid)
+        run_dir = os.path.join(wandb_dir, run_path)
+        files_dir = os.path.join(run_dir, "files")
+        sync_file = os.path.join(run_dir, "run-{}.wandb".format(start_time))
+        os.makedirs(files_dir)
+        settings = dict(
+            log_internal=os.path.join(run_dir, "internal.log"),
+            files_dir=files_dir,
+            _start_time=start_time,
+            _start_datetime=start_datetime,
+            disable_code=None,
+            code_program=None,
+            save_code=None,
+            sync_file=sync_file,
+            _internal_queue_timeout=20,
+            _internal_check_process=8,
+            _disable_meta=True,
+            _disable_stats=False,
+            git_remote=None,
+            program=None,
+            resume=None,
+            ignore_globs=(),
+            offline=None,
+            _log_level=log_level,
+            run_id=None,
+            entity=None,
+            project=None,
+            run_group=None,
+            run_job_type=None,
+            run_tags=None,
+            run_name=None,
+            run_notes=None,
+            _jupyter=None,
+            _kaggle=None,
+            _offline=None,
+            email=None,
+            silent=None,
+        )
```

Comment:
Should we put these defaults somewhere central?  I recall seeing other places where we constructing an object like this.

Reply:
It was in wandb sync code, but i think that has been improved,  i added a note to use a real settings object here also